File: easy school/MDIParent1.vb
Imports System.Windows.Forms
Public Class MDIParent1
    Public role As String

    Private Sub ShowNewForm(ByVal sender As Object, ByVal e As EventArgs)
        ' Create a new instance of the child form.
        Dim ChildForm As New System.Windows.Forms.Form
        ' Make it a child of this MDI form before showing it.
        ChildForm.MdiParent = Me

        m_ChildFormNumber += 1
        ChildForm.Text = "Window " & m_ChildFormNumber

        ChildForm.Show()

    End Sub

    Private Sub OpenFile(ByVal sender As Object, ByVal e As EventArgs)
        Dim OpenFileDialog As New OpenFileDialog
        OpenFileDialog.InitialDirectory = My.Computer.FileSystem.SpecialDirectories.MyDocuments
        OpenFileDialog.Filter = "Text Files (*.txt)|*.txt|All Files (*.*)|*.*"
        If (OpenFileDialog.ShowDialog(Me) = System.Windows.Forms.DialogResult.OK) Then
            Dim FileName As String = OpenFileDialog.FileName
            ' TODO: Add code here to open the file.
        End If
    End Sub

    Private Sub SaveAsToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Dim SaveFileDialog As New SaveFileDialog
        SaveFileDialog.InitialDirectory = My.Computer.FileSystem.SpecialDirectories.MyDocuments
        SaveFileDialog.Filter = "Text Files (*.txt)|*.txt|All Files (*.*)|*.*"

        If (SaveFileDialog.ShowDialog(Me) = System.Windows.Forms.DialogResult.OK) Then
            Dim FileName As String = SaveFileDialog.FileName
            ' TODO: Add code here to save the current contents of the form to a file.
        End If
    End Sub


    Private Sub ExitToolsStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Me.Close()
    End Sub

    Private Sub CutToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        ' Use My.Computer.Clipboard to insert the selected text or images into the clipboard
    End Sub

    Private Sub CopyToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        ' Use My.Computer.Clipboard to insert the selected text or images into the clipboard
    End Sub

    Private Sub PasteToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        'Use My.Computer.Clipboard.GetText() or My.Computer.Clipboard.GetData to retrieve information from the clipboard.
    End Sub



    Private Sub CascadeToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Me.LayoutMdi(MdiLayout.Cascade)
    End Sub

    Private Sub TileVerticalToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Me.LayoutMdi(MdiLayout.TileVertical)
    End Sub

    Private Sub TileHorizontalToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Me.LayoutMdi(MdiLayout.TileHorizontal)
    End Sub

    Private Sub ArrangeIconsToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Me.LayoutMdi(MdiLayout.ArrangeIcons)
    End Sub

    Private Sub CloseAllToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        ' Close all child forms of the parent.
        For Each ChildForm As Form In Me.MdiChildren
            ChildForm.Close()
        Next
    End Sub

    Private m_ChildFormNumber As Integer

    Private Sub MDIParent1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        If role = 1 Then
            AdminToolStripMenuItem.Enabled = True
        End If
        Dim ChildForm As New view_all
        closeall()
        With ChildForm
            .MdiParent = Me
            ' .ControlBox = False
            '.WindowState = FormWindowState.Maximized
            .Dock = DockStyle.Fill
            .Text = ""
            .Show()
        End With
    End Sub

    Private Sub PrintToolStripButton_Click(sender As Object, e As EventArgs)

    End Sub

    Private Sub StudentsToolStripMenuItem_Click(sender As Object, e As EventArgs)
        ' Dim ChildForm As New register_main
        closeall()
        '' Make it a child of this MDI form before showing it.
        'ChildForm.MdiParent = Me
        'ChildForm.Show()
    End Sub
    Sub closeall()
        For Each ChildForm As Form In Me.MdiChildren
            ChildForm.Close()
        Next
    End Sub

    Private Sub RegisterNewToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles RegisterNewToolStripMenuItem.Click
        Dim bb As New new_student
        bb.ShowDialog()
    End Sub

    Private Sub ModifyToolStripMenuItem_Click(sender As Object, e As EventArgs)
        'Dim ChildForm As New student_profile
        'closeall()
        '' Make it a child of this MDI form before showing it.
        'With ChildForm
        '    .MdiParent = Me
        '    .ControlBox = False
        '    .WindowState = FormWindowState.Maximized
        '    .Text = ""
        '    .Show()
        'End With
    End Sub

    Private Sub ViewRecordsToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles ViewRecordsToolStripMenuItem.Click
        Dim ChildForm As New view_all
        closeall()
        With ChildForm
            .MdiParent = Me
            ' .ControlBox = False
            '.WindowState = FormWindowState.Maximized
            .Dock = DockStyle.Fill
            .Text = ""
            .Show()
        End With
    End Sub

    Private Sub RegisterToolStripMenuItem_Click(sender As Object, e As EventArgs)
        'Dim ChildForm As New teacher
        'closeall()
        '' Make it a child of this MDI form before showing it.
        'With ChildForm
        '    .MdiParent = Me
        '    .ControlBox = False
        '    .WindowState = FormWindowState.Maximized
        '    .Text = ""
        '    .Show()
        'End With
    End Sub

    Private Sub ViewRecordsToolStripMenuItem1_Click(sender As Object, e As EventArgs)
        'Dim ChildForm As New teacher_view
        'closeall()
        '' Make it a child of this MDI form before showing it.
        'With ChildForm
        '    .MdiParent = Me
        '    .ControlBox = False
        '    .WindowState = FormWindowState.Maximized
        '    .Text = ""
        '    .Show()
        'End With
    End Sub

    Private Sub AssignClassTeacherToolStripMenuItem_Click(sender As Object, e As EventArgs)
        'Dim ChildForm As New class_assgn
        'closeall()
        '' Make it a child of this MDI form before showing it.
        'With ChildForm
        '    .MdiParent = Me
        '    .ControlBox = False
        '    .WindowState = FormWindowState.Maximized
        '    .Text = ""
        '    .Show()
        'End With
    End Sub

    Private Sub AssignASubjectToolStripMenuItem_Click(sender As Object, e As EventArgs)
    End Sub

    Private Sub CreateSubjectsToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles CreateSubjectsToolStripMenuItem.Click
        Process.Start("sms.exe")
    End Sub

    Private Sub SchoolFeesToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles SchoolFeesToolStripMenuItem.Click
        Dim ChildForm As New Fees_Admin
        closeall()
        With ChildForm
            .MdiParent = Me
            ' .ControlBox = False
            '.WindowState = FormWindowState.Maximized
            .Dock = DockStyle.Fill
            .Text = ""
            .Show()
        End With
        'view_all
    End Sub

    Private Sub FeesReportsToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles FeesReportsToolStripMenuItem.Click
        Dim ChildForm As New report
        closeall()
        With ChildForm
            .MdiParent = Me
            ' .ControlBox = False
            '.WindowState = FormWindowState.Maximized
            .Dock = DockStyle.Fill
            .Text = ""
            .Show()
        End With
    End Sub

    Private Sub AddUsersToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles AddUsersToolStripMenuItem.Click
        Dim ChildForm As New create_user
        '  closeall()
        With ChildForm
            .ShowDialog()

        End With
    End Sub

    Private Sub ViewProjectionsToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles ViewProjectionsToolStripMenuItem.Click
        Dim ChildForm As New billing
        closeall()
        With ChildForm
            .MdiParent = Me
            .Dock = DockStyle.Fill
            .Text = ""
            .Show()
        End With
    End Sub


    Private Sub ClassesAndStreamsToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles ClassesAndStreamsToolStripMenuItem.Click
        Dim ChildForm As New classes
        '  closeall()
        With ChildForm
            .ShowDialog()
        End With
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs)
        Me.Close()

    End Sub

    Private Sub AddNewTeacherToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles AddNewTeacherToolStripMenuItem.Click
        Dim ChildForm As New teacher
        closeall()
        With ChildForm
            '.MdiParent = Me
            '.Dock = DockStyle.Fill
            .Text = ""
            .ShowDialog()
        End With
    End Sub

    Private Sub ClassTeachersToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles ClassTeachersToolStripMenuItem.Click
        Dim ChildForm As New class_teachers

        With ChildForm
            '.MdiParent = Me
            '.Dock = DockStyle.Fill
            .Text = ""
            .ShowDialog()
        End With
    End Sub

    Private Sub ViewTeachersToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles ViewTeachersToolStripMenuItem.Click
        Dim ChildForm As New teacher_view
        closeall()
        With ChildForm
            .MdiParent = Me
            .Dock = DockStyle.Fill
            .Text = ""
            .Show()
        End With
    End Sub

    Private Sub TEachersAttendanceToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles TEachersAttendanceToolStripMenuItem.Click
        Dim ChildForm As New teachers_attendance

        With ChildForm
            '.MdiParent = Me
            '.Dock = DockStyle.Fill
            .Text = ""
            .ShowDialog()
        End With
    End Sub

    Private Sub SubjectDesignationToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles SubjectDesignationToolStripMenuItem.Click
        Dim ChildForm As New class_assgn

        With ChildForm
            '.MdiParent = Me
            '.Dock = DockStyle.Fill
            .Text = ""
            .ShowDialog()
        End With
    End Sub

    Private Sub RegisterExamResultsToolStripMenuItem_Click(sender As Object, e As EventArgs)
        Dim ChildForm As New Form2
        closeall()
        With ChildForm
            .MdiParent = Me
            .Dock = DockStyle.Fill
            .Text = ""
            .Show()
        End With
    End Sub

    Private Sub ViewExamResultsToolStripMenuItem_Click(sender As Object, e As EventArgs)
        '    Dim ChildForm As New 
        '    closeall()
        '    With ChildForm
        '        .MdiParent = Me
        '        .Dock = DockStyle.Fill
        '        .Text = ""
        '        .Show()
        '    End With
    End Sub

    Private Sub PrintExamReportsToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles PrintExamReportsToolStripMenuItem.Click
        Dim ChildForm As New exam_reports
        closeall()
        With ChildForm
            .Text = ""
            .ShowDialog()
        End With
    End Sub
End Class

File: easy school.ConvertedToC#/MDIParent1.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
using System.Threading.Tasks;
namespace easy_school
{
	public partial class MDIParent1
	{

		public string role;
		private void ShowNewForm(object sender, EventArgs e)
		{
			// Create a new instance of the child form.
			System.Windows.Forms.Form ChildForm = new System.Windows.Forms.Form();
			// Make it a child of this MDI form before showing it.
			ChildForm.MdiParent = this;

			m_ChildFormNumber += 1;
			ChildForm.Text = "Window " + m_ChildFormNumber;

			ChildForm.Show();

		}

		private void OpenFile(object sender, EventArgs e)
		{
			OpenFileDialog OpenFileDialog = new OpenFileDialog();
			OpenFileDialog.InitialDirectory = easy_school.My.MyProject.Computer.FileSystem.SpecialDirectories.MyDocuments;
			OpenFileDialog.Filter = "Text Files (*.txt)|*.txt|All Files (*.*)|*.*";
			if ((OpenFileDialog.ShowDialog(this) == System.Windows.Forms.DialogResult.OK)) {
				string FileName = OpenFileDialog.FileName;
				// TODO: Add code here to open the file.
			}
		}

		private void SaveAsToolStripMenuItem_Click(object sender, EventArgs e)
		{
			SaveFileDialog SaveFileDialog = new SaveFileDialog();
			SaveFileDialog.InitialDirectory = easy_school.My.MyProject.Computer.FileSystem.SpecialDirectories.MyDocuments;
			SaveFileDialog.Filter = "Text Files (*.txt)|*.txt|All Files (*.*)|*.*";

			if ((SaveFileDialog.ShowDialog(this) == System.Windows.Forms.DialogResult.OK)) {
				string FileName = SaveFileDialog.FileName;
				// TODO: Add code here to save the current contents of the form to a file.
			}
		}


		private void ExitToolsStripMenuItem_Click(object sender, EventArgs e)
		{
			this.Close();
		}

		private void CutToolStripMenuItem_Click(object sender, EventArgs e)
		{
			// Use My.Computer.Clipboard to insert the selected text or images into the clipboard
		}

		private void CopyToolStripMenuItem_Click(object sender, EventArgs e)
		{
			// Use My.Computer.Clipboard to insert the selected text or images into the clipboard
		}

		private void PasteToolStripMenuItem_Click(object sender, EventArgs e)
		{
			//Use My.Computer.Clipboard.GetText() or My.Computer.Clipboard.GetData to retrieve information from the clipboard.
		}



		private void CascadeToolStripMenuItem_Click(object sender, EventArgs e)
		{
			this.LayoutMdi(MdiLayout.Cascade);
		}

		private void TileVerticalToolStripMenuItem_Click(object sender, EventArgs e)
		{
			this.LayoutMdi(MdiLayout.TileVertical);
		}

		private void TileHorizontalToolStripMenuItem_Click(object sender, EventArgs e)
		{
			this.LayoutMdi(MdiLayout.TileHorizontal);
		}

		private void ArrangeIconsToolStripMenuItem_Click(object sender, EventArgs e)
		{
			this.LayoutMdi(MdiLayout.ArrangeIcons);
		}

		private void CloseAllToolStripMenuItem_Click(object sender, EventArgs e)
		{
			// Close all child forms of the parent.
			foreach (Form ChildForm in this.MdiChildren) {
				ChildForm.Close();
			}
		}


		private int m_ChildFormNumber;
		private void MDIParent1_Load(object sender, EventArgs e)
		{
			if (role == 1) {
				AdminToolStripMenuItem.Enabled = true;
			}
			view_all ChildForm = new view_all();
			closeall();
			var _with1 = ChildForm;
			_with1.MdiParent = this;
			// .ControlBox = False
			//.WindowState = FormWindowState.Maximized
			_with1.Dock = DockStyle.Fill;
			_with1.Text = "";
			_with1.Show();
		}


		private void PrintToolStripButton_Click(object sender, EventArgs e)
		{
		}

		private void StudentsToolStripMenuItem_Click(object sender, EventArgs e)
		{
			// Dim ChildForm As New register_main
			closeall();
			//' Make it a child of this MDI form before showing it.
			//ChildForm.MdiParent = Me
			//ChildForm.Show()
		}
		public void closeall()
		{
			foreach (Form ChildForm in this.MdiChildren) {
				ChildForm.Close();
			}
		}

		private void RegisterNewToolStripMenuItem_Click(object sender, EventArgs e)
		{
			new_student bb = new new_student();
			bb.ShowDialog();
		}

		private void ModifyToolStripMenuItem_Click(object sender, EventArgs e)
		{
			//Dim ChildForm As New student_profile
			//closeall()
			//' Make it a child of this MDI form before showing it.
			//With ChildForm
			//    .MdiParent = Me
			//    .ControlBox = False
			//    .WindowState = FormWindowState.Maximized
			//    .Text = ""
			//    .Show()
			//End With
		}

		private void ViewRecordsToolStripMenuItem_Click(object sender, EventArgs e)
		{
			view_all ChildForm = new view_all();
			closeall();
			var _with2 = ChildForm;
			_with2.MdiParent = this;
			// .ControlBox = False
			//.WindowState = FormWindowState.Maximized
			_with2.Dock = DockStyle.Fill;
			_with2.Text = "";
			_with2.Show();
		}

		private void RegisterToolStripMenuItem_Click(object sender, EventArgs e)
		{
			//Dim ChildForm As New teacher
			//closeall()
			//' Make it a child of this MDI form before showing it.
			//With ChildForm
			//    .MdiParent = Me
			//    .ControlBox = False
			//    .WindowState = FormWindowState.Maximized
			//    .Text = ""
			//    .Show()
			//End With
		}

		private void ViewRecordsToolStripMenuItem1_Click(object sender, EventArgs e)
		{
			//Dim ChildForm As New teacher_view
			//closeall()
			//' Make it a child of this MDI form before showing it.
			//With ChildForm
			//    .MdiParent = Me
			//    .ControlBox = False
			//    .WindowState = FormWindowState.Maximized
			//    .Text = ""
			//    .Show()
			//End With
		}

		private void AssignClassTeacherToolStripMenuItem_Click(object sender, EventArgs e)
		{
			//Dim ChildForm As New class_assgn
			//closeall()
			//' Make it a child of this MDI form before showing it.
			//With ChildForm
			//    .MdiParent = Me
			//    .ControlBox = False
			//    .WindowState = FormWindowState.Maximized
			//    .Text = ""
			//    .Show()
			//End With
		}

		private void AssignASubjectToolStripMenuItem_Click(object sender, EventArgs e)
		{
		}

		private void CreateSubjectsToolStripMenuItem_Click(object sender, EventArgs e)
		{
			Process.Start("sms.exe");
		}

		private void SchoolFeesToolStripMenuItem_Click(object sender, EventArgs e)
		{
			Fees_Admin ChildForm = new Fees_Admin();
			closeall();
			var _with3 = ChildForm;
			_with3.MdiParent = this;
			// .ControlBox = False
			//.WindowState = FormWindowState.Maximized
			_with3.Dock = DockStyle.Fill;
			_with3.Text = "";
			_with3.Show();
			//view_all
		}

		private void FeesReportsToolStripMenuItem_Click(object sender, EventArgs e)
		{
			report ChildForm = new report();
			closeall();
			var _with4 = ChildForm;
			_with4.MdiParent = this;
			// .ControlBox = False
			//.WindowState = FormWindowState.Maximized
			_with4.Dock = DockStyle.Fill;
			_with4.Text = "";
			_with4.Show();
		}

		private void AddUsersToolStripMenuItem_Click(object sender, EventArgs e)
		{
			create_user ChildForm = new create_user();
			//  closeall()
			var _with5 = ChildForm;
			_with5.ShowDialog();

		}

		private void ViewProjectionsToolStripMenuItem_Click(object sender, EventArgs e)
		{
			billing ChildForm = new billing();
			closeall();
			var _with6 = ChildForm;
			_with6.MdiParent = this;
			_with6.Dock = DockStyle.Fill;
			_with6.Text = "";
			_with6.Show();
		}


		private void ClassesAndStreamsToolStripMenuItem_Click(object sender, EventArgs e)
		{
			classes ChildForm = new classes();
			//  closeall()
			var _with7 = ChildForm;
			_with7.ShowDialog();
		}

		private void Button1_Click(object sender, EventArgs e)
		{
			this.Close();

		}

		private void AddNewTeacherToolStripMenuItem_Click(object sender, EventArgs e)
		{
			teacher ChildForm = new teacher();
			closeall();
			var _with8 = ChildForm;
			//.MdiParent = Me
			//.Dock = DockStyle.Fill
			_with8.Text = "";
			_with8.ShowDialog();
		}

		private void ClassTeachersToolStripMenuItem_Click(object sender, EventArgs e)
		{
			class_teachers ChildForm = new class_teachers();

			var _with9 = ChildForm;
			//.MdiParent = Me
			//.Dock = DockStyle.Fill
			_with9.Text = "";
			_with9.ShowDialog();
		}

		private void ViewTeachersToolStripMenuItem_Click(object sender, EventArgs e)
		{
			teacher_view ChildForm = new teacher_view();
			closeall();
			var _with10 = ChildForm;
			_with10.MdiParent = this;
			_with10.Dock = DockStyle.Fill;
			_with10.Text = "";
			_with10.Show();
		}

		private void TEachersAttendanceToolStripMenuItem_Click(object sender, EventArgs e)
		{
			teachers_attendance ChildForm = new teachers_attendance();

			var _with11 = ChildForm;
			//.MdiParent = Me
			//.Dock = DockStyle.Fill
			_with11.Text = "";
			_with11.ShowDialog();
		}

		private void SubjectDesignationToolStripMenuItem_Click(object sender, EventArgs e)
		{
			class_assgn ChildForm = new class_assgn();

			var _with12 = ChildForm;
			//.MdiParent = Me
			//.Dock = DockStyle.Fill
			_with12.Text = "";
			_with12.ShowDialog();
		}

		private void RegisterExamResultsToolStripMenuItem_Click(object sender, EventArgs e)
		{
			Form2 ChildForm = new Form2();
			closeall();
			var _with13 = ChildForm;
			_with13.MdiParent = this;
			_with13.Dock = DockStyle.Fill;
			_with13.Text = "";
			_with13.Show();
		}

		private void ViewExamResultsToolStripMenuItem_Click(object sender, EventArgs e)
		{
			//    Dim ChildForm As New 
			//    closeall()
			//    With ChildForm
			//        .MdiParent = Me
			//        .Dock = DockStyle.Fill
			//        .Text = ""
			//        .Show()
			//    End With
		}

		private void PrintExamReportsToolStripMenuItem_Click(object sender, EventArgs e)
		{
			exam_reports ChildForm = new exam_reports();
			closeall();
			var _with14 = ChildForm;
			_with14.Text = "";
			_with14.ShowDialog();
		}
		public MDIParent1()
		{
			Load += MDIParent1_Load;
			InitializeComponent();
		}
	}
}
